client/question/services/TipsGeneratorService.js

```diff
@@ -24,31 +24,56 @@ tie.factory('TipsGeneratorService', [
       FeedbackParagraphObjectFactory) {
     /**
      * Returns a list of paragraphs indicating system-generated tips for the
-     * learner.
+     * learner. At most one system-generated tip is presented.
      *
      * @param {string} language The language in which the code is written.
-     * @param {string} code The code to analyze.
-     * @returns {Array} A list of FeedbackParagraph objects, representing
-     *   system-generated tips, to prepend to the regular TIE feedback.
+     * @param {string} codeLines The lines of code to analyze.
```

Comment:
Is this a string or an array? Used as an array below?

Reply:
Done.